### PR TITLE
[core][sql] enum SPELLTYPE and spell_list.sql enum Implementation

### PR DESCRIPTION
**Pull request by DotesMiqotes**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Implements `enum SPELLTYPE` to clean up and optimize the `bool is()` functions in `spell.cpp`.
Implements enum style variables in `spell_list.sql` for the `group` `family` `element` `validTargets` `skill` and `type` columns.
Adds a `type` column to `spell_list.sql`.
Adds `setSpellType()` to `spell.cpp` to load the spell types from the database.
Adds `getSpellType()` to `spell.cpp` that returns the spell type flags.
Aranges the columns of `spell_list.sql` into a more readable format.
Fixes every instance of a spell having the wrong element, valid targets, or skill.
Adds all necessary types to all ninjutsu spells and bard songs to ensure they sort properly into their respective mob spell containers.

## Steps to test these changes

I compiled the server, and used `source sql/spell_list.sql` to update the database. I then play tested the changes by running around and fighting as many types of casters I could in Castle O with a party full of trusts. Trusts, mobs I was fighting, idle mobs, and their pets were all casting everything they should have been as expected.
I'm using the same testing method after upgrading `spell_list.sql` and everything is working as expected. Mobs are buffing themselves, nuking and debuffing me as well as my trusts when I engage them. My trusts are all casting the spells they should be, and I'm having no issues with any of my spells on RNG/RDM.

### CODE REVIEW

**Review by zach2good — COMMENTED** [review]
Can spells have multiple `SpellTypes`? Otherwise, there's no need to store/use this as a flag and check everything as `spellType & SPELLTYPE_HEAL`. I'd argue it's cleaner and easier to understand in the `spell->isHeal()` form.

Does the current implementation cover all spells of all types?

Just as a point of styling that isn't written down anywhere: it doesn't feel so nice to put this new numeric type column in the table at the end of the row, where a few other tables keep the last entry for the expansion string. It also winds up being another magic number that you have to go back and forth to know/remember what it is. I think you can see in the items tables we've started using aliases to make these sort of things more human readable.

**Comment by DotesMiqotes** [comment]
Under the current logic, they're calculated as if they have multiple types.

This covers all spells in the database, I went through line by line making sure they had every type the server was calculating them as. I also ninja fixed the skill of Gravity II (it wasn't enfeebling).

I debated putting the `type` column either just before or after the `skill` column, but it would have made it even less readable than it is now. I felt that `content_tag` acts as a nice reference point in the row to count out from. It definitely made things a bit easier on me when I was determining types. Moving it later is a fairly easy task if needed.

YAML databases would also be great for static stuff like this. They're meant to be human readable. From what I remember working with them on rAthena, they were a bit more efficient than SQL as well.

**Comment by DotesMiqotes** [comment]
Quick update, I'm going to be converting more than the new magic numbers to aliases.

**Comment by Xaver-DaRed** [comment]
Being honest, theres sonething about this that just feels wrong to me.
For starters the enums themselves. Spells could fall into many categories, sure, for instance, healing spells CAN be damage spells.

However, this enum list doesnt have a damage category, nor an aborb category (drain, aspir, etc) nor many others.
Probably for good reaso: they arent used nor needed in the code.
So that leads me to believe that this categories are just minor exceptions, and dont really require this rework.
And that leads to another thing: The database.

I can only speak for myself, but, I personally want to see less static data in the database, not more.
The reason for this is that, the more data theres in the database, the less readable it is and the less likely ppl areto come to help fix it.

In short: My real question here would be "Why is this being done?" or "Whats the issue we are trying to fix with this?"

No offense intended, just my opinion and doubts on the matter.

**Comment by DotesMiqotes** [comment]
The logic is predetermined and types don't need to be recalculated every time a spell is cast, or added to a spell list container. This change collapses multiple if statements into one every time the spell type flag is used. I didn't add anything for damage or drains because in the cases a type is used for them, they're sorted into a catch all. My other ninja fix involved that, the logic determined that all debuffs were being sorted into a mob's damage spell container, and the debuff container was staying empty. Those 'special cases' are most of `spell_list.sql`.

I already have the entire type column converted to aliases now as well, it's done wonders for readability. I plan on doing the same with a few of the other columns before I make the commit.

This was one of the improvements that I found I could make when I was doing the research needed to address this:
https://github.com/LandSandBoat/server/blob/be17f7f3c3edfe40e12576854da0a17c2bf2c481/scripts/actions/spells/trust/joachim.lua#L30

Update: I finished the `group` column. This part is definitely worth it. I have a few more columns to go, and I plan on sliding the `type` column over on my last pass.
Update2: Done with `family` and about halfway through `element` I've been correcting spell elements as I go.
Update3: `element` is done, and I lost count of how many I fixed. It's technically perfect now. I have at least Valid Targets, and Skills to go.
Update4: `validTargets` and `skill` are done now, `zonemisc` is mostly 0s with a few 128s and 8s for certain things, Sliding that to the end of the aliases, and moving `type` over to the alias block seem to the be the last significant things I can do at this point.
```sql
INSERT INTO `spell_list` VALUES (20,'cursna',0x00001D00000000000000000000000000000000200000,@G_WHITE,@F_NA,@E_LIGHT,0,@VT_SUPPSPL,@S_HEALING,30,1000,10000,0,0,20,2000,4,0,1.00,1,165,4,204,NULL,@T_HMNA);
INSERT INTO `spell_list` VALUES (21,'holy',0x00003200000037000000000000000000000000000000,@G_WHITE,@F_HOLY,@E_LIGHT,0,@VT_ENEMY,@S_DIVINE,100,750,60000,2,252,21,2000,0,125,1.00,0,0,0,204,NULL,@T_NONE);
INSERT INTO `spell_list` VALUES (22,'holy_ii',0x00005F00000063000000000000000000000000000000,@G_WHITE,@F_HOLY,@E_LIGHT,0,@VT_ENEMY,@S_DIVINE,150,1250,60000,2,252,22,2000,0,250,1.00,0,0,0,204,'ABYSSEA',@T_NONE);
INSERT INTO `spell_list` VALUES (23,'dia',0x00000300010000000000000000000000000000000000,@G_WHITE,@F_DIA,@E_LIGHT,0,@VT_ENEMY,@S_ENFEEBLING,7,1000,5000,2,252,23,2000,0,2,1.00,0,0,0,204,NULL,@T_DEBUFF);
```
Update5: The bulk of the work is done, I'm going to add proper types for ninjutsu spells since it effects how they're sorted into mob spell containers, as well as update types for other spells I fixed, and commented stuff I tracked down information on.
```sql
INSERT INTO `spell_list` VALUES (57,'haste',0x00002800300000000000000000000000000000000000,@G_WHITE,@F_HASTE,@E_WIND,@VT_SUPPSPL,@S_ENHANCING,@T_BUFF,0,40,3000,20000,230,0,57,2000,0,0,1.00,1,165,0,204,NULL);
INSERT INTO `spell_list` VALUES (58,'paralyze',0x00000400060000000000000000000000000000000000,@G_WHITE,@F_PARALYZE,@E_ICE,@VT_ENEMY,@S_ENFEEBLING,@T_DEBUFF,0,6,3000,10000,84,84,58,2000,0,0,1.00,1,176,0,204,NULL);
INSERT INTO `spell_list` VALUES (59,'silence',0x00000F00120000000000000000000000000000000000,@G_WHITE,@F_NONE,@E_WIND,@VT_ENEMY,@S_ENFEEBLING,@T_DEBUFF,0,16,3000,10000,0,0,59,2000,0,0,1.00,1,176,0,204,NULL);
INSERT INTO `spell_list` VALUES (60,'barfire',0x00000000110000000000000000000000000000000010,@G_WHITE,@F_ELE_BAR,@E_WATER,@VT_SELF,@S_ENHANCING,@T_BUFF,0,6,500,10000,230,0,60,2000,4,0,1.00,1,165,0,0,NULL);
```

File: src/map/spell.h [hunk 1]
```diff
@@ -234,6 +234,18 @@ enum SPELLFLAG
...
+{
+    SPELLTYPE_NONE   = 0x00,
+    SPELLTYPE_HEAL   = 0x01,
+    SPELLTYPE_CURE   = 0x02,
+    SPELLTYPE_BUFF   = 0x04,
+    SPELLTYPE_DEBUFF = 0x08,
+    SPELLTYPE_NA     = 0x10,
+    SPELLTYPE_RAISE  = 0x20,
+    SPELLTYPE_SEVERE = 0x40,
+};
```
Comment (Xaver-DaRed) [inline src/map/spell.h:247]: Whats the difference between heal and cure?

File: src/map/spell.cpp [hunk 2]
```diff
@@ -452,7 +424,7 @@ namespace spell
     void LoadSpellList()
     {
         const char* Query = "SELECT spellid, name, jobs, `group`, family, validTargets, skill, castTime, recastTime, animation, animationTime, mpCost, \
-                             AOE, base, element, zonemisc, multiplier, message, magicBurstMessage, CE, VE, requirements, content_tag, spell_range \
+                             AOE, base, element, zonemisc, multiplier, message, magicBurstMessage, CE, VE, requirements, content_tag, type, spell_range \
```
Comment (Xaver-DaRed) [inline src/map/spell.cpp:427]: While we are here, why dont we order this properly, following the same order as the database?

File: src/map/spell.h (hunk 1 above)
Comment (Xaver-DaRed) [reply src/map/spell.h:247]: Heal is healing magic, in general, ok

Cure is cure and cura, families... Why not use families then?

File: src/map/spell.cpp (hunk 2 above)
Comment (DotesMiqotes) [reply src/map/spell.cpp:427]: Sounds good. I'll make it happen.

File: src/map/spell.h (hunk 1 above)
Comment (DotesMiqotes) [reply src/map/spell.h:247]: Types are based on this. On their own `SPELLTYPE_CURE` and `SPELLTYPE_HEAL` won't get used much if ever, Valid targets are taken into consideration when determining the actual spell type. Cure, Curaga, and Cura also 3 separate families.
```c++
bool CSpell::isHeal()
{
    return ((getValidTarget() & TARGET_SELF) && getSkillType() == SKILL_HEALING_MAGIC) || m_ID == SpellID::Pollen || m_ID == SpellID::Wild_Carrot ||
           m_ID == SpellID::Healing_Breeze || m_ID == SpellID::Magic_Fruit;
}
```
```c++
bool CSpell::isCure()
{
    return ((static_cast<uint16>(m_ID) >= 1 && static_cast<uint16>(m_ID) <= 11) || m_ID == SpellID::Cura || m_ID == SpellID::Cura_II ||
            m_ID == SpellID::Cura_III);
}
```
```c++
bool CSpell::isBuff() const
{
    return (getValidTarget() & TARGET_SELF) && !(getValidTarget() & TARGET_ENEMY);
}
```
```
Cure   = @G_WHITE,@F_CURE  ,@E_LIGHT,@VT_CURESPL ,@S_HEALING,@T_HMCURE
Curaga = @G_WHITE,@F_CURAGA,@E_LIGHT,@VT_SLF_PRTY,@S_HEALING,@T_BUFFHMCURE
Cura   = @G_WHITE,@F_CURA  ,@E_LIGHT,@VT_SELF    ,@S_HEALING,@T_BUFFHMCURE
```

File: src/map/spell.cpp (hunk 2 above)
Comment (DotesMiqotes) [reply src/map/spell.cpp:427]:
```sql
INSERT INTO `spell_list` VALUES (1,'cure',@G_WHITE,@F_CURE,@E_LIGHT,@VT_CURESPL,@S_HEALING,@T_HMCURE,0,8,2000,5000,7,7,1,2000,4,0,1.00,0,0,0,204,@CT_BASE,'WHM:1,RDM:3,PLD:5,SCH:5');
```

Comment (DotesMiqotes) [reply src/map/spell.cpp:427]: RIP `requirements`
```sql
INSERT INTO `spell_list` VALUES (16,'blindna',@G_WHITE,@F_NA,@E_LIGHT,@VT_SUPPSPL,@S_HEALING,@T_HMNA,0,16,1000,10000,0,0,16,2000,4,0,1.00,1,165,4,204,@CT_BASE,'WHM:14,SCH:17:AW');
```

Comment (Xaver-DaRed) [reply src/map/spell.cpp:427]: I will say, if this grows too big, it will be a nightmare to review properly.

Maybe the simpler changes should go in separate PRs first.

Comment (DotesMiqotes) [reply src/map/spell.cpp:427]: There won't be anything difficult to review by the time I'm done. I already reordered everything in the SQL query, and arranged everything that comes after before starting this change. It won't take me much to rework the logic required to read the database.
```sql
INSERT INTO `spell_list` VALUES (11,'curaga_v',@G_WHITE,@F_CURAGA,@E_LIGHT,@VT_SLF_PRTY,@S_HEALING,@T_BUFFHMCURE,0,380,5750,11000,0,0,11,2000,1,0,1.00,0,0,204,@CT_ABY,'WHM:91');
INSERT INTO `spell_list` VALUES (12,'raise',@G_WHITE,@F_RAISE,@E_LIGHT,@VT_PDEAD,@S_HEALING,@T_RAISE,0,150,15000,60000,0,0,12,2000,0,0,1.00,0,0,204,@CT_BASE,'WHM:25,RDM:38,PLD:50,SCH:35');
INSERT INTO `spell_list` VALUES (13,'raise_ii',@G_WHITE,@F_RAISE,@E_LIGHT,@VT_PDEAD,@S_HEALING,@T_RAISE,0,150,14000,60000,0,0,13,2000,0,0,1.00,0,0,204,@CT_BASE,'WHM:56,RDM:95,SCH:70:AW'); -- req 4
```

Comment (DotesMiqotes) [reply src/map/spell.cpp:427]: If it makes you feel any better about the scope of this, I'm pulling all the job and level data off of the scrolls at the auction house to put in the new col.
```sql
INSERT INTO `spell_list` VALUES (152,'blizzard_iv',@G_BLACK,@F_BLIZZARD,@E_ICE,@VT_ENEMY,@S_ELEMENTAL,@T_NONE,0,162,6000,30000,2,252,152,2000,0,506,2.00,0,0,204,@CT_BASE,'BLM:74,RDM:89,SCH:74:AB,GEO:88'); -- req 2
INSERT INTO `spell_list` VALUES (153,'blizzard_v',@G_BLACK,@F_BLIZZARD,@E_ICE,@VT_ENEMY,@S_ELEMENTAL,@T_NONE,0,267,10000,45000,2,252,153,2000,0,829,2.30,0,0,204,@CT_ABY,'BLM:89,SCH:95:AB'); -- req 2
INSERT INTO `spell_list` VALUES (154,'aero',@G_BLACK,@F_AERO,@E_WIND,@VT_ENEMY,@S_ELEMENTAL,@T_NONE,0,6,500,2000,2,252,154,2000,0,25,1.00,0,0,204,@CT_BASE,'BLM:9,RDM:14,DRK:17,SCH:12,GEO:14');
```
for example
req 1 looks like merit ability from what I can see, could easily be "WHM:75:M"
req 2 is Addendum: Black "SCH:99:AB"
req 4 is Addendum: White "SCH:99:AW"
req 8 is Tabula Rasa, and I have that for Embrava now "SCH:5:TR"
req 16 is for the latest blu stuff
req 32 is for geo stuff

**Comment by DotesMiqotes** [comment]
Still a WIP but I wanted to give an update on where I'm at.
```sql
CREATE TABLE `spell_list` (
  `id` smallint(3) unsigned NOT NULL,
  `name` varchar(24) NOT NULL,
  `group` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `family` smallint(4) unsigned NOT NULL DEFAULT '0',
  `element` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `validTargets` smallint(3) unsigned NOT NULL DEFAULT '0',
  `skill` tinyint(2) unsigned NOT NULL DEFAULT '0',
  `type` tinyint(2) unsigned NOT NULL DEFAULT '0',
  `mpCost` smallint(4) unsigned NOT NULL DEFAULT '0',
  `baseDMG` smallint(4) unsigned NOT NULL DEFAULT '0',
  `multiplier` float(4,2) NOT NULL DEFAULT '1.00',
  `CE` int(10) unsigned NOT NULL DEFAULT '0',
  `VE` int(10) unsigned NOT NULL DEFAULT '0',
  `castRange` smallint(3) unsigned NOT NULL DEFAULT '0',
  `castTime` smallint(5) unsigned NOT NULL DEFAULT '0', -- TOENUM
  `recastTime` int(10) unsigned NOT NULL DEFAULT '0', -- TOENUM
  `message` smallint(5) unsigned NOT NULL DEFAULT '0', -- IN PROCESS ENUM
  `magicBurstMessage` smallint(5) NOT NULL DEFAULT '0', -- TOENUM
  `animation` smallint(5) unsigned NOT NULL DEFAULT '0',
  `animationTime` smallint(4) unsigned NOT NULL DEFAULT '2000',
  `AOE` tinyint(1) unsigned NOT NULL DEFAULT '0', -- TOENUM
  `content_tag` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `requirements` varchar(50) NOT NULL DEFAULT '',
  PRIMARY KEY (`id`)
)
```
```c++

    uint16     getID() const;
//--std:string getName() const;
    uint8      getGroup() const;
    uint16     getFamily() const;
    uint16     getElement() const;
    uint16     getValidTargets() const;
    uint8      getSkill() const;
    uint8      getType() const;
    uint16     getMPCost() const;
    uint16     getBaseDMG() const;
    float      getMultiplier() const;
    uint16     getCE() const;
    uint16     getVE() const;
    float      getCastRange() const;
    uint32     getCastTime() const;
    uint32     getRecastTime() const;
    uint16     getMessage() const;
    uint16     getMagicBurstMessage() const;
    uint16     getAnimation() const;
    uint16     getAnimationTime() const;
    uint8      getAOE() const;
    uint8      getContentTag() const;

    uint8      getJob(JOBTYPE JobID) const; // TODO
    uint8      getRequirements() const;     // TODO
    uint16     getZoneMisc() const;         // TODO
```
Mostly relaxing and reorganizing things right now, I'll update with a few lines of the DB after I reorder them.

Update:
Looks like *this pass through this file is going to be my last. -- I can't english rn
```sql
INSERT INTO `spell_list` VALUES (1,'cure',@G_WHITE,@F_CURE,@E_LIGHT,@VT_CURESPL,@S_HEALING,@T_HMCURE,8,0,1.00,0,0,@R_20,@C_2s,@RC_5s,@M_7,@MB_7,1,@AT_2s,@AOE_4,@CT_BASE,'WHM:1,RDM:3,PLD:5,SCH:5');
INSERT INTO `spell_list` VALUES (2,'cure_ii',@G_WHITE,@F_CURE,@E_LIGHT,@VT_CURESPL,@S_HEALING,@T_HMCURE,24,0,1.00,0,0,@R_20,@C_2s_25,@RC_5s_5,@M_7,@MB_7,2,@AT_2s,@AOE_4,@CT_BASE,'WHM:11,RDM:14,PLD:17,SCH:17');
INSERT INTO `spell_list` VALUES (3,'cure_iii',@G_WHITE,@F_CURE,@E_LIGHT,@VT_CURESPL,@S_HEALING,@T_HMCURE,46,0,1.00,0,0,@R_20,@C_2s_5,@RC_6s,@M_7,@MB_7,3,@AT_2s,@AOE_4,@CT_BASE,'WHM:21,RDM:26,PLD:30,SCH:30');
INSERT INTO `spell_list` VALUES (4,'cure_iv',@G_WHITE,@F_CURE,@E_LIGHT,@VT_CURESPL,@S_HEALING,@T_HMCURE,88,0,1.00,0,0,@R_20,@C_2s_5,@RC_8s,@M_7,@MB_7,4,@AT_2s,@AOE_4,@CT_BASE,'WHM:41,RDM:48,PLD:55,SCH:55');
INSERT INTO `spell_list` VALUES (5,'cure_v',@G_WHITE,@F_CURE,@E_LIGHT,@VT_CURESPL,@S_HEALING,@T_HMCURE,135,0,1.00,0,0,@R_20,@C_2s_5,@RC_10s,@M_7,@MB_7,5,@AT_2s,@AOE_0,@CT_BASE,'WHM:61');
INSERT INTO `spell_list` VALUES (6,'cure_vi',@G_WHITE,@F_CURE,@E_LIGHT,@VT_CURESPL,@S_HEALING,@T_HMCURE,227,0,1.00,0,0,@R_20,@C_3s_25,@RC_15s,@M_7,@MB_7,6,@AT_2s,@AOE_0,@CT_ABY,'WHM:80');
```
Update2:
I got this fully updated with everything I could see in `settings/map.lua`. I'll clean up the comments when I hook it up everywhere.
```sql
-- Content Tag
SET @CT_REST = 0;  -- Restrict Content
SET @CT_BASE = 1;  -- Base Game
SET @CT_ROZ  = 2;  -- Rise of the Zilart
SET @CT_COP  = 3;  -- Chains of Promathia
SET @CT_TOAU = 4;  -- Treasures of Aht Urzzz... ~ sp
SET @CT_WOTG = 5;  -- Wings of the Goddess
SET @CT_ACP  = 6;  -- A Crystalline Prophecy
SET @CT_AMK  = 7;  -- A Moogle de whatever ~ sp
SET @CT_ASA  = 8;  -- A Shantotto Ascension
SET @CT_ABY  = 9;  -- Abyssemeh ~ sp
SET @CT_SOA  = 10; -- Sons of Anarchy ~ sp
SET @CT_ROV  = 11; -- Cool one with trusts ~ sp
SET @CT_TVR  = 12; -- Kinda sucked in Gran Truismo ~ sp
SET @CT_VOID = 13; -- Voidwatch ~ Not Implemented
```
Update3:
I realized that if I space the cols right, a simple find `" ,@G_"` and replace `"  ,@G_"` would make it easy to instantly add or remove an extra space through the entire file as needed in a matter of seconds. I'll be including it in the final pass.
```sql
INSERT INTO `spell_list` VALUES (   1,'cure'      ,@G_WHITE ,@F_CURE ,@E_LIGHT ,@VT_CURESPL ,@S_HEALING ,@T_HMCURE,   8, 0, 1.00, 0, 0 ,@R_20 ,@C_2s    ,@RC_5s   ,@M_7 ,@MB_7,  1 ,@AT_2s ,@AOE_4 ,@CT_BASE ,'WHM:1 ,RDM:3 ,PLD:5 ,SCH:5 ');
INSERT INTO `spell_list` VALUES (   2,'cure_ii'   ,@G_WHITE ,@F_CURE ,@E_LIGHT ,@VT_CURESPL ,@S_HEALING ,@T_HMCURE,  24, 0, 1.00, 0, 0 ,@R_20 ,@C_2s_25 ,@RC_5s_5 ,@M_7 ,@MB_7,  2 ,@AT_2s ,@AOE_4 ,@CT_BASE ,'WHM:11,RDM:14,PLD:17,SCH:17');
INSERT INTO `spell_list` VALUES (   3,'cure_iii'  ,@G_WHITE ,@F_CURE ,@E_LIGHT ,@VT_CURESPL ,@S_HEALING ,@T_HMCURE,  46, 0, 1.00, 0, 0 ,@R_20 ,@C_2s_5  ,@RC_6s   ,@M_7 ,@MB_7,  3 ,@AT_2s ,@AOE_4 ,@CT_BASE ,'WHM:21,RDM:26,PLD:30,SCH:30');
INSERT INTO `spell_list` VALUES (   4,'cure_iv'   ,@G_WHITE ,@F_CURE ,@E_LIGHT ,@VT_CURESPL ,@S_HEALING ,@T_HMCURE,  88, 0, 1.00, 0, 0 ,@R_20 ,@C_2s_5  ,@RC_8s   ,@M_7 ,@MB_7,  4 ,@AT_2s ,@AOE_4 ,@CT_BASE ,'WHM:41,RDM:48,PLD:55,SCH:55');
INSERT INTO `spell_list` VALUES (   5,'cure_v'    ,@G_WHITE ,@F_CURE ,@E_LIGHT ,@VT_CURESPL ,@S_HEALING ,@T_HMCURE, 135, 0, 1.00, 0, 0 ,@R_20 ,@C_2s_5  ,@RC_10s  ,@M_7 ,@MB_7,  5 ,@AT_2s ,@AOE_0 ,@CT_BASE ,'WHM:61');
INSERT INTO `spell_list` VALUES (   6,'cure_vi'   ,@G_WHITE ,@F_CURE ,@E_LIGHT ,@VT_CURESPL ,@S_HEALING ,@T_HMCURE, 227, 0, 1.00, 0, 0 ,@R_20 ,@C_3s_25 ,@RC_15s  ,@M_7 ,@MB_7,  6 ,@AT_2s ,@AOE_0 ,@CT_ABY  ,'WHM:80');
```
Update4:
There is logic in the core to determine this, it should be set in the database. I'll create another col and add it.
```c++
    SPELLFLAG_NONE           = 0x00,
    SPELLFLAG_HIT_ALL        = 0x01, // Hit all targets in range regardless of party
    SPELLFLAG_WIPE_SHADOWS   = 0x02, // Wipe shadows even if single target and miss/resist (example: "Maiden's Virelai")
    SPELLFLAG_IGNORE_SHADOWS = 0x04, // Ignore shadows and hit player anyways (example: Mobs "Death" spell)
```
Already done, it took 2 seconds to run. I'll be including it in the final pass, working on column spacing from the left since that's all set now.

**Comment by DotesMiqotes** [comment]
Please merge my PRs.

**Comment by DotesMiqotes** [comment]
Good luck with things.

**Comment by zach2good** [comment]
As a note for your future endeavours with open-source software, crowd-sourced development, and working with distributed groups of people: "Please merge my PRs" will not make maintainers swarm to review and merge your PRs.

All the best out there.
